sqld/src/replication/frame.rs

```diff
@@ -43,64 +51,130 @@ impl fmt::Debug for Frame {
     }
 }
 
-impl Frame {
-    /// size of a single frame
-    pub const SIZE: usize = size_of::<FrameHeader>() + WAL_PAGE_SIZE as usize;
+/// Owned version of a frame, on the heap
```

Comment:
I'm missing the purpose of the new FrameMut and FrameBorrowed. Why do we need it? Feel free to also post the answer in the commit message

Reply:
sorry, only got back to this PR now. So basically, `FrameBorrowed` is a stack data structure that contains the frame's raw data. But since it's quite big, you don't really ever get an owned version. Instead, you get a `Frame` and a `FrameMut` that are the stack-allocated equivalents. Both deref to a FrameBorrowed, with the difference between `FrameMut` and `Frame` being that `Frame` is a shared pointer that is cheaply clonable, and `FrameMut` is an exclusive reference to the Frame.